src/conversion.rs

```diff
@@ -244,25 +244,77 @@ where
     }
 }
 
-/// Extract reference to instance from `PyObject`
-impl<'a, T> FromPyObject<'a> for &'a T
-where
-    T: PyTryFrom<'a>,
-{
-    #[inline]
-    fn extract(ob: &'a PyAny) -> PyResult<&'a T> {
-        Ok(T::try_from(ob)?)
+pub mod extract_impl {
+    use super::*;
+
+    pub trait ExtractImpl<'a, Target>: Sized {
+        fn extract(source: &'a PyAny) -> PyResult<Target>;
+    }
+
+    pub struct Cloned;
+    pub struct Reference;
+    pub struct MutReference;
```

Comment:
Do we need `MutReference`?
We can also do
```rust

    impl<'a, T> ExtractImpl<'a, &'a mut T> for Reference
    where
        T: PyTryFrom<'a>,
    {
        fn extract(source: &'a PyAny) -> PyResult<&'a mut T> {
            Ok(T::try_from_mut(source)?)
        }
    }
```
...
But, it is also good to leave it as is for ergonomics.

Reply:
I think it's clearer to have all three, even if we only technically need two.